pkg/router/routeapihelpers/validation.go

```diff
@@ -281,6 +290,27 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 	return result
 }
 
+// validatePEMContent validates that the passed content, when PEM decoded contains
+// the expected type. This validation is necessary to block attempts of passing certificates
+// inside the field that should be private key only, and vice versa. This is required
+// because otherwise passing an out of order certificate to HAProxy will break it.
+// The function will take the array of bytes of a key/cert, and the expected type
+// that the PEM should content. The match here is not strict, but will check if the
+// passed byte type CONTAINS the expected pemType
+func validatePEMContent(content []byte, pemType string) error {
+	for {
+		var pemBlock *pem.Block
+		pemBlock, content = pem.Decode(content)
```

Comment:
Does this need a check, just in case the content is empty?
```suggestion
        if len(content) > 0 {
		pemBlock, content = pem.Decode(content)
```

Reply:
ah actually, this is inside a "for" that keeps decoding the content and reducing the slice, so at some point content will be 0, and the error will always happen if we add this validation. 

The right fix here will be `for len(content) > 0 { ....`

Reply:
Except that means that empty content is valid, because it would return nil.